src/Microsoft.TemplateEngine.Cli/Installer.cs

```diff
@@ -63,6 +63,55 @@ public void InstallPackages(IEnumerable<string> installationRequests, IList<stri
                 }
             }
 
+            if (localSources.Count > 0)
+            {
+                InstallLocalPackages(localSources.Select(x => new InstallationRequest(x)), debugAllowDevInstall);
+            }
+
+            if (packages.Count > 0)
+            {
+                IList<string> additionalDotNetArguments = (interactive) ? new string[] { "--interactive" } : null;
+                InstallRemotePackages(packages, nuGetSources, additionalDotNetArguments, false);
+            }
+
+            _environmentSettings.SettingsLoader.Save();
+        }
+
+        public void InstallPackages(IEnumerable<InstallationRequest> installationRequests, IList<string> nuGetSources = null, bool debugAllowDevInstall = false, bool interactive = false)
+        {
+            List<InstallationRequest> localSources = new List<InstallationRequest>();
+            List<Package> packages = new List<Package>();
+            List<Package> optionalPackages = new List<Package>();
```

Comment:
I settled on `packagesFromOptionalWorkloads` if that makes sense.

Reply:
On second thought, I'm wondering if we should be using the word "workload" at all (applies to the entire PR). The API where the information about these packages comes from is named `GetDotnetSdkTemplatePackages()` and it returns a list of `IOptionalSdkTemplatePackageInfo`'s. We're introducing the word "workload" here and it may not be appropriate.

How about `optionalSdkPackages`?

Reply:
In that case, would `IsPartOfAnOptionalWorkload` need to be renamed as well (defined in `IInstallUnitDescriptor` and its implementors )?

Reply:
Yes. But up to you. It's a cosmetic change and can be done later.

Reply:
I will merge & close this to unblock others on optional workloads for the time being. Let's consider this in the next refactoring chance we get.